.github/workflows/validate-workflows.yml

```diff
@@ -0,0 +1,17 @@
+name: "Validate Workflows"
+
+on:
+  push:
+  pull_request:
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  validate-workflows:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Ensure GitHub actions are SHA pinned
+        uses: zgosalvez/github-actions-ensure-sha-pinned-actions@5d6ac37a4cef8b8df67f482a8e384987766f0213 # v3.0.17
```

Comment:
I'm hesitant on using this action but it doesn't seem like there are (m)any alternatives. Have you audited the code?

Reply:
I think we might need some rules for external actions, in this case the source is very basic. I'll rewrite it as bash script.